[24기  지의선] Add: 로그인 페이지 완료

esphoenixc opened the pull request:
-login page function added
-main page layout still in progress
-메인 페이지 레이아웃 짜는중

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 버튼 활성화 기능 구현(예시)
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화(예시)

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [ ] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [ ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ ] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

joonsikyang commented:
라벨 표시도 해주세요!

joonsikyang commented on students/11th/SangwookRyu/js/login.js:
```diff
@@ -11,27 +11,3 @@ buttonOnOffPw.addEventListener('keyup', function(e) {
     btn.disabled = true;
   }
```
깔끔하게 잘 하셨네요! 삼항연산자 적용하는 방식으로 바꿔보시기 바랍니다.
- https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

ryuinkyoto replied:
if else문 삼항연산자로 바꿨습니다

joonsikyang commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,13 @@
+const buttonOnOffComment = document.getElementById('writecomment');
+
+buttonOnOffComment.addEventListener('keyup', function(e) {
+  const comment = document.getElementById('writecomment').value;
+  let btn = document.getElementById('addcomment');
```
const, let 구분을 어떻게 하고 계신지 설명 부탁드립니다 :)

ryuinkyoto replied:
const는 바뀌지 않을 값에만 사용하고있습니다. let btn도 const로 바꿨습니다.

joonsikyang commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,13 @@
+  const comment = document.getElementById('writecomment').value;
+  let btn = document.getElementById('addcomment');
+  
+  if (comment.length > 0) {
+    btn.disabled = false;
+  }
+  else {
+    btn.disabled = true;
+  }
```
마찬가지로 삼항연산자 적용 부탁드립니다.

ryuinkyoto replied:
적용했습니다

joonsikyang commented on students/11th/SangwookRyu/main.html:
```diff
@@ -8,57 +8,154 @@
-          </div>
-          <div class="users-comment"></div>
+        <div class="btnbox">
+          <div class="btn-explore"><img alt="explore"src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/explore.png"></div>
+          <div class="btn-heart"><img alt="heart"src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png"></div>
+          <div class="btn-profile"><img alt="profile"src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/profile.png"></div>
```
부모 자식 관계가 잘 드러날 수 있도록 img 태그 한 줄 아래로 내려주시고 들여쓰기 해주세요.

ryuinkyoto replied:
적용했습니다

joonsikyang commented on students/11th/SangwookRyu/main.html:
```diff
@@ -8,57 +8,154 @@
+          <div class="copyright">© 2020 INSTAGRAM FROM FACEBOOK</div>
+        </footer>
+      </aside>
+    </section>
```
1) class명에 camelCase가 안 보이네요! convention 잘 지켜주세요.
- https://github.com/wecode-bootcamp-korea/westagram-frontend/issues/26
- 변수명, 함수명 짓는 부분 참고해주세요.

2) id 보다는 class를 우선적으로 사용해주세요. 나중에 리액트로 바꾸는 경우에도 그렇고 id 사용은 최대한 지양해주는 것이 좋습니다. 참고하세요!

ryuinkyoto replied:
1. html class 명에도 camelCase를 사용해도되나요? 예를들면 loginBox보단 login-box가 css에는 더 적합하지 않나요?

2. id 전부 class로 바꿨습니다

joonsikyang commented on students/11th/SangwookRyu/style/common.css:
```diff
@@ -1,9 +1,34 @@
+  outline: none;
+}
+
+html, body {
```
html 태그에 스타일을 지정해주신 이유가 있을까요?

ryuinkyoto replied:
테스트 하고 지우는걸 깜빡했습니다

joonsikyang commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,13 @@
+const buttonOnOffComment = document.getElementById('writecomment');
+
+buttonOnOffComment.addEventListener('keyup', function(e) {
```
이벤트로 인자를 받을 필요가 있는 경우인가요?

ryuinkyoto replied:
인자 받을필요 없는거 확인하고 삭제했습니다

jamessoun93 commented:
@joonsikyang @saengmotmi @dipsiiiiiiiiii 
force-push 제가했어요 :)

ryuinkyoto commented:
> @joonsikyang @saengmotmi @dipsiiiiiiiiii
> force-push 제가했어요 :)

File Changed 목록이 엄청 지저분해졌는데 괜찮은가요...?

saengmotmi commented on students/11th/SangwookRyu/main.html:
```diff
@@ -0,0 +1,165 @@
+     <link href="style/main.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <!--navigation-->
```
주석 제거해주세요

ryuinkyoto replied:
넵

saengmotmi commented on students/11th/SangwookRyu/main.html:
```diff
@@ -0,0 +1,165 @@
+        <div class="logo-box">
+          <div class="logo1"><img alt = "icon-instagram" src ="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/logo.png"></div>
+          <div class="line"></div>
+          <div class="logo2"><img alt = "logo-instagram" src ="/img/logo_text.png"></div>
```
부모 자식 관계가 잘 드러날 수 있도록 줄바꿈 해주세요
```suggestion
          <div class="logo2">
            <img alt = "logo-instagram" src ="/img/logo_text.png">
          </div>
```

ryuinkyoto replied:
수정했습니다

saengmotmi commented on students/11th/SangwookRyu/main.html:
```diff
@@ -0,0 +1,165 @@
+          <img alt="photo" src="/students/11th/SangwookRyu/img/summer.jpg" >
+        </div>
+        <div class="article-btn">
+          <img alt="heart" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png">
```
alt 속성 굿 👍

saengmotmi commented on students/11th/SangwookRyu/style/main.css:
```diff
@@ -0,0 +1,325 @@
+
+/* article */
+.main-box {
+    display: flex;
+    flex-direction: row;
```
```suggestion
    display: flex;
    flex-direction: row;
```
flex를 쓰는 순간 가로로 배열되지 않나요?

ryuinkyoto replied:
flex-direction: row 지웠습니다

ryuinkyoto commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,10 @@
+const btnOnOffComment = document.getElementsByClassName('write-comment')[0];
+
+btnOnOffComment.addEventListener('keyup', function() {
+  const comment = document.getElementsByClassName('write-comment')[0].value;
```
변수 사용했습니다

ryuinkyoto commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,10 @@
+  const btn = document.getElementsByClassName('add-comment')[0];
+
+  btn.disabled = comment.length > 0 ? false : true;
+  console.log("hello");
```
지웠습니다

ryuinkyoto commented on students/11th/SangwookRyu/login.html:
```diff
@@ -0,0 +1,24 @@
+          <input class="password" type="password" placeholder="비밀번호">
+          <a href="main.html"><button class="login-button" disabled>로그인</button></a>
+        </div>
+      <div class="forget-pw">
```
넵

ryuinkyoto commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,10 @@
+  const comment = document.getElementsByClassName('write-comment')[0].value;
+  const btn = document.getElementsByClassName('add-comment')[0];
+
+  btn.disabled = comment.length > 0 ? false : true;
```
false와 true를 지우게되면

  if (e.keyCode === 13 && btn.disabled ===false) {

이부분이 작동하지 않게되는데 다른 방법이 있나요?

joonsikyang replied:
해당 내용(`if (e.keyCode === 13 && btn.disabled ===false)`)은 아래에 있는 내용이네요. 위의 경우처럼 입력된 값을 비교해서 버튼이 활성화 되는 경우는 이해가 되는데 아래 함수에서도 삼항연산자로 작성해주신 이유가 있을까요?
댓글이 입력되고 나서 input value를 빈 스트링 값으로 바꿔주는 것 처럼 `btn.disabled = false;`로 해주면 될 것 같습니다.

joonsikyang commented on students/11th/SangwookRyu/js/login.js:
```diff
@@ -0,0 +1,10 @@
+btnOnOffPw.addEventListener('keyup', function() {
+  const id = document.getElementsByClassName('login-id')[0].value;
+  const pw = document.getElementsByClassName('password')[0].value;
+  let btn = document.getElementsByClassName('login-button')[0];
```
const로 변수 선언해주세요.

joonsikyang commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,56 @@
+
+const commentBtnClick = document.getElementsByClassName('add-comment')[0];
+commentBtnClick.addEventListener('click', function (){
+  const commentValue = document.getElementsByClassName('write-comment')[0];
+  const commentBox = document.getElementsByClassName('comment-box')[0];
+  const commentUl = document.createElement('ul');
+  const commentTitle = document.createElement('li');
+  const commentText = document.createElement('li');
+  const btn = document.getElementsByClassName('add-comment')[0];
+
+  commentUl.className = "comment-text";
+  commentTitle.innerHTML = "ryuvVv";
+  commentTitle.className = "viewer-id";
+  commentText.innerHTML = commentValue.value;
+  commentText.className = "viewer-comment";
+  commentUl.appendChild(commentTitle);
+  commentUl.appendChild(commentText);
+  commentBox.appendChild(commentUl);
```
- 아이디와 댓글을 모두 `<li>` 태그로 하신 이유가 있을까요? 둘을 `<li>` 태그로 감싸고 각각은 `<span>` 태그 혹은 `<div>` 태그를 사용하는 것이 좋을 것 같습니다.
- innerHTML 메소드와 템플릿 리터럴을 활용하면 코드 양을 좀 더 줄여볼 수 있을 것 같습니다. 참고해주세요.

```js
commentUl.innerHTML = `
  <li>
    <span class="viewer-id">ryuvVv</span>
    <div class="viewer-comment">${commentValue}</div>
  </li>
`
```

joonsikyang replied:
- 아이디와 댓글을 모두 `<li>` 태그로 하신 이유가 있을까요? 둘을 `<li>` 태그로 감싸고 각각은 `<span>` 태그 혹은 `<div>` 태그를 사용하는 것이 좋을 것 같습니다.

- innerHTML 메소드와 템플릿 리터럴을 활용하면 코드 양을 좀 더 줄여볼 수 있을 것 같습니다. 참고해주세요.

```js
commentUl.innerHTML = `
  <li>
    <span class="viewer-id">ryuvVv</span>
    <div class="viewer-comment">${commentValue.value}</div>
  </li>
`
```

joonsikyang commented on students/11th/SangwookRyu/js/main.js:
```diff
@@ -0,0 +1,56 @@
+const commentBtnEnter = document.getElementsByClassName('write-comment')[0];
+
+commentBtnEnter.addEventListener('keypress', function(e) {
+  const commentValue = document.getElementsByClassName('write-comment')[0];
+  const commentBox = document.getElementsByClassName('comment-box')[0];
+  const commentUl = document.createElement('ul');
+  const commentTitle = document.createElement('li');
+  const commentText = document.createElement('li');
+  const btn = document.getElementsByClassName('add-comment')[0];
+
+  if (e.keyCode === 13 && btn.disabled ===false) {
+  commentUl.className = "comment-text";
+  commentTitle.innerHTML = "ryuvVv";
+  commentTitle.className = "viewer-id";
+  commentText.innerHTML = commentValue.value;
+  commentText.className = "viewer-comment";
+  commentUl.appendChild(commentTitle);
+  commentUl.appendChild(commentText);
+  commentBox.appendChild(commentUl);
+
+  btn.disabled = commentValue.length > 0 ? false : true;
+  commentValue.value = "";
```
위 내용과 함수가 비슷하네요. handleComment 함수를 정의해서 그 안에 함수 내용을 담고, 조건문 통과 시에 해당 함수가 실행되도록 하면 좋을 것 같습니다.

Commit:
Add: 로그인 페이지 완료
-login page function added
-main page layout still in progress
-메인 페이지 레이아웃 짜는중

esphoenixc committed:
Delete mainprac.html

esphoenixc committed:
Delete mainprac1.css

esphoenixc committed:
Delete prac2.css

esphoenixc committed:
Delete login.js

esphoenixc committed:
Delete login.html

esphoenixc committed:
Delete main.html

esphoenixc committed:
Delete main.css

esphoenixc committed:
Delete prac2.html

esphoenixc committed:
Delete prac1.html

yeonuk-hwang committed:
Docs: guide/main.md 파일 수정

- 필요한 이미지에 만료된 이미지 URL 삭제
- 피드 사진, 프로필 사진은 가지고 있는 이미지 활용 또는 무료 이미지 사이트 활용하도록 변경

Commit:
ADD: 로그인 페이지 완성
     로그인페이지 아이디 비밀번호 입력시 로그인 버튼 활성화 완성
     메인페이지 레이아웃 완성
     로그인 페이지 아이디 비밀번호 입력시 로그인버튼 누른뒤 alert msg 추가

In Progress:
            -메인페이지
            -댓글추가 기능

Commit:
merge local and remote repository

Merge branch 'feature/euiseon24' of https://github.com/wecode-bootcamp-korea/westagram-frontend into feature/euiseon24
